lib/idp_common_pkg/idp_common/summarization/service.py

```diff
@@ -398,13 +398,22 @@ def process_document(
         from idp_common.utils import normalize_boolean_value
 
         enabled = normalize_boolean_value(summarization_config.get("enabled", True))
+        output_bucket = os.environ["OUTPUT_BUCKET"]
         if not enabled:
             logger.info(
                 f"Summarization is disabled in configuration for document {document.id}, skipping processing"
             )
             # Update document status to completed if not already failed
             if document.status != Status.FAILED:
                 document.status = Status.COMPLETED
+            try:
+                # When we Reprocess a document and don't enable Summarization we lose the summary in the UI
```

Comment:
I do not know how 'reprocessing a document' has anything to do with the scope of the PR, which was simply (as the title suggests) to make Classification and Extraction optional.. Simple, safe, uncomplicated..    
This seems to have evolved into something much more, which, while it may be great (if I understood it :) ) , is riskier and not the 'quick win' I'd hoped for.